`katib-db-manager` unit tests are failing

Running `tox -ve unit` for `katib-db-manager` fails with the following error message:

```
  File "/home/ubuntu/katib-operators/charms/katib-db-manager/tests/unit/test_operator.py", line 172, in test_update_status
    assert harness.charm.model.unit.status == charm_status
AssertionError: assert BlockedStatus('Please add required database relation: eg. relational-db') == ActiveStatus('')
 +  where BlockedStatus('Please add required database relation: eg. relational-db') = <ops.model.Unit katib-db-manager/0>.status
 +    where <ops.model.Unit katib-db-manager/0> = <ops.model.Model object at 0xffffae468880>.unit
 +      where <ops.model.Model object at 0xffffae468880> = <ops.testing.Harness.begin.<locals>.TestCharm object at 0xffffae523790>.model
 +        where <ops.testing.Harness.begin.<locals>.TestCharm object at 0xffffae523790> = <ops.testing.Harness object at 0xffffae440700>.charm
------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------
ERROR    charm:charm.py:423 Failed to handle <ConfigChangedEvent via KatibDBManagerOperator/on/config_changed[3]> with error: Please add required database relation: eg. relational-db
ERROR    charm:charm.py:423 Failed to handle <PebbleReadyEvent via KatibDBManagerOperator/on/katib_db_manager_pebble_ready[5]> with error: Please add required database relation: eg. relational-db
ERROR    charm:charm.py:423 Failed to handle <PebbleReadyEvent via KatibDBManagerOperator/on/katib_db_manager_pebble_ready[6]> with error: Please add required database relation: eg. relational-db
ERROR    charm:charm.py:423 Failed to handle <UpdateStatusEvent via KatibDBManagerOperator/on/update_status[7]> with error: Please add required database relation: eg. relational-db
================================================================================== short test summary info ===================================================================================
FAILED tests/unit/test_operator.py::test_update_status[CheckStatus.UP-charm_status0] - AssertionError: assert BlockedStatus('Please add required database relation: eg. relational-db') == ActiveStatus('')
================================================================================ 1 failed, 10 passed in 0.21s ================================================================================
ERROR: InvocationError for command /home/ubuntu/katib-operators/charms/katib-db-manager/.tox/unit/bin/coverage run --source=/home/ubuntu/katib-operators/charms/katib-db-manager/src/ -m pytest --ignore=/home/ubuntu/katib-operators/charms/katib-db-manager/tests/integration -vv --tb native (exited with code 1)
```

This issue is seen locally running `tox -ve unit`, it seems like we are not testing this in the GitHub CI.

--- a/.github/workflows/integrate.yaml
+++ b/.github/workflows/integrate.yaml
@@ -42,6 +42,19 @@ jobs:
     - name: Check flake8
       run: flake8 ./charms/*/src
 
+  unit:
+    name: Unit tests
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        charm:
+          - katib-db-manager
+    steps:
+      - uses: actions/checkout@v3
+      - run: python3 -m pip install tox
+      - run: tox -e ${{ matrix.charm }}-unit
+  
   charm-integration:
     name: Integration tests (microk8s)
     runs-on: ubuntu-20.04

--- a/charms/katib-db-manager/tests/unit/test_operator.py
+++ b/charms/katib-db-manager/tests/unit/test_operator.py
@@ -161,8 +161,22 @@ def test_update_status(
     Test update status handler.
     Check on the correct charm status when health check status is UP/DOWN.
     """
+    database = MagicMock()
+    fetch_relation_data = MagicMock()
+    fetch_relation_data.return_value = {
+        "test-db-data": {
+            "endpoints": "host:1234",
+            "username": "username",
+            "password": "password",
+        }
+    }
+    database.fetch_relation_data = fetch_relation_data
+    harness.model.get_relation = MagicMock(
+        side_effect=_get_relation_db_only_side_effect_func
+    )
     harness.set_leader(True)
     harness.begin_with_initial_hooks()
+    harness.charm.database = database
     harness.container_pebble_ready("katib-db-manager")
 
     _get_check_status.return_value = health_check_status

--- a/tox.ini
+++ b/tox.ini
@@ -7,8 +7,7 @@ max-line-length = 100
 [tox]
 skipsdist = True
 skip_missing_interpreters = True
-envlist = fmt, lint, unit, integration, {katib-controller}-{lint,unit,integration}
-
+envlist = fmt, lint, unit, integration, {katib-controller,katib-db-manager}-{lint,unit,integration}
 [vars]
 all_path = {[vars]tst_path}
 tst_path = {toxinidir}/tests/

